### ISSUE TITLE
regression loss 0

### ISSUE THREAD

**Issue by cdebom** [fizyr/keras-retinanet#276]
Hi, I am trying to test and learn on the retinanet.
I am testing the retinanet  (two classes) with 2000 x 2000  pixels images with 10K small  objects ~30 x 30.

but when I try
retinanet-train csv trainretina.txt clsmap.txt
I am getting a 0 regression loss:

1/10000 [..............................] - ETA: 43:24:45 - loss: 0.1008 - regression_loss: 0.0000e+00 - classification_loss: 0.10082018-02-07 21:06:15.531919: I tensorflow/core/common_runtime/gpu/pool_allocator.cc:247] PoolAllocator: After 1885 get requests, put_count=1718 evicted_count=1000 eviction_rate=0.582072 and unsatisfied allocation rate=0.672149
2018-02-07 21:06:15.532004: I tensorflow/core/common_runtime/gpu/pool_allocator.cc:259] Raising pool_size_limit_ from 100 to 110

    2/10000 [..............................] - ETA: 22:00:08 - loss: 0.1008 - regression_loss: 0.0000e+00 - classification_loss: 0.1008
    3/10000 [..............................] - ETA: 14:50:26 - loss: 0.1007 - regression_loss: 0.0000e+00 - classification_loss: 0.1007
    4/10000 [..............................] - ETA: 11:15:35 - loss: 0.1007 - regression_loss: 0.0000e+00 - classification_loss: 0.1007
    5/10000 [..............................] - ETA: 9:06:45 - loss: 0.1007 - regression_loss: 0.0000e+00 - classification_loss: 0.1007 

Does anybody has an idea on the issue I am getting here? My guess is that it should have to do with the fact that my images are big and objects small. 

Thanks

**Comment by yhenon** [comment]
>My guess is that it should have to do with the fact that my images are big and objects small.

This is almost certainly the case. Look at the `Debugging` section on the Readme to see how to check if your boxes are too small. Given 2000px images and 30px objects, this is nearly certain.

I'm closing this since it is extremely likely the issue is objects are too small. You will need to cut up your images before passing them to the training network.

**Comment by cdebom** [comment]
Thanks for the advise. I will try to look at the debugging. However, for the application I have in mind this is pretty much always the case, i.e., I will have lots of big images full of tiny objects. Would you have any advise to adapt the code to such case?

**Comment by hgaiser** [comment]
At those resolutions and object sizes you will definitely run into issues.
If you allow the generator to downsize less (currently it downsizes your
images to 600x600), you will probably run into memory issues before your
objects get big enough. My advice: cut up your images into smaller parts,
possible with an overlap to avoid objects on the border.
